platform/linux-generic/odp_packet_io.c

```diff
@@ -2065,6 +2065,77 @@ int odp_pktin_queue_config(odp_pktio_t pktio,
 	return 0;
 }
 
+int _odp_pktio_pktout_tm_config(odp_pktio_t pktio_hdl,
+				odp_pktout_queue_t *queue, bool reconf)
+{
+	odp_pktout_queue_param_t param;
+	bool pktio_started = false;
+	odp_pktout_mode_t mode;
+	pktio_entry_t *entry;
+	unsigned int i;
+	int rc = 0;
+
+	odp_pktout_queue_param_init(&param);
+	param.num_queues = 1;
+
+	entry = get_pktio_entry(pktio_hdl);
+	if (entry == NULL) {
+		ODP_DBG("pktio entry %d does not exist\n", pktio_hdl);
+		return -1;
+	}
+
+	rc = -ENOTSUP;
+	mode = entry->s.param.out_mode;
+	/* Don't proceed further if mode is not TM */
+	if (mode != ODP_PKTOUT_MODE_TM)
+		return rc;
+
+	/* Don't reconfigure unless requested */
+	if (entry->s.num_out_queue && !reconf) {
```

Comment:
Here we have a race between multiple threads creating TM systems at the same time on top of the same pktio egress. I guess a new lock around this queue creation code here would take care of it.

Reply:
The existing tm_glb->create_lock in the caller can probably handle this if the pktout config is moved below it, right?

Reply:
Ya, thats true as for PKTIO in TM mode, everyone can only configure pktout using _odp_pktio_pktout_tm_config() via tm_create(). Will move it down in v3